collections/map.go

```diff
@@ -20,6 +20,11 @@ type Map[K, V any] struct {
 	sa     func(context.Context) store.KVStore
 	prefix []byte
 	name   string
+
+	// isSecondaryIndex indicates that this map represents a secondary index
+	// on another collection and that it should be skipped when generating
+	// a user facing schema
+	isSecondaryIndex bool
```

Comment:
q: could it be useful to still generate this optionally? Or is it absolutely unnecessary? (asking out of ignorance)

Reply:
The target database (ex. postgres) should generally be able to produce any secondary index it wants if the primary data is present. So this isn't needed.